### DISCUSSION TITLE
Change language

### DISCUSSION THREAD

**Discussion by ghost** [tsujan/FeatherPad#622] [Q&A] [answered]
Hello,

I am trying to test my translation of the help file. 
After a successful build from source (following INSTALL instructions) under Debian 32 bit laptop with locale pt_PT.UTF-8, FeatherPad always starts with English GUI. Curiously enough, the help file is called via menu with the pt_PT translation.
How can I correct that, please?

**Comment by tsujan** [comment]
First, thanks for contributing to translations!

Sorry, I've no idea why localization doesn't work in your system. Here, when I use FeatherPad with pt_PT.UTF-8, it starts with the Portuguese translation. Does localization work with other *Qt apps* in your system?

I converted your report to a Discussion for 2 reasons: it wasn't an issue in FeatherPad; and others might find your question more easily here and they might have a clue.

**Reply by tsujan** [reply]
Did you install `qttools5-dev-tools` before compilation?

**Comment by ghost** [comment]
Yes, it is installed. I tried smplayer, which is also a qt app, and it works fine. Under preferences, interface, the selection is system language, pt_PT.UTF-8.
Is there any option to 'force' a language on compilation? Maybe the files are not being updated.

**Reply by tsujan** [reply]
> Is there any option to 'force' a language on compilation?

No. If you have `qttools5-dev-tools`, the translation files are created and will be installed with `sudo make install`.

FeatherPad with pt_PT.UTF-8:

![fpad](https://user-images.githubusercontent.com/981076/151981160-d4e3456c-d85f-4c9e-945b-e2f528e356a6.png)

Also, check that you haven't installed it in `/usr/local/...`, although it isn't likely.

**Comment by ghost** [comment]
/usr/local/share/featherpad has the help files.
/usr/local/bin/ has both featherpad and the fpad link.
Nothing else. Can you please tell me exactly where the translation files should be, after installation?

**Reply by tsujan** [reply]
Translations are installed in `${CMAKE_INSTALL_PREFIX}/share/featherpad/translations`. In your case, `CMAKE_INSTALL_PREFIX` is `/usr/local`, which isn't usual on Debian. Debian installs everything in `/usr`.

Installing FeatherPad (or any other app) in `/usr/local` on Debian could cause trouble later. If you install the same app by using `apt` later, there will be two versions of it but the old one inside `/usr/local` will have priority, so that the new one will be made hidden to the user.

**Comment by ghost** [comment, marked as answer]
That did it: `cmake CMAKE_INSTALL_PREFIX=/usr ..`
Thank you for your help.

**Reply by tsujan** [reply]
> That did it: cmake CMAKE_INSTALL_PREFIX=/usr ..

Good.

This is a strange behavior of Debian. In Arch and its derivatives, there's no need to set `CMAKE_INSTALL_PREFIX` explicitly.

Since, apparently, you didn't make a deb package but installed it directly, make sure none of its files remains in `/usr/local`.

**Reply by ghost** [reply]
Maybe not so strange. I followed an [online tutorial](https://sharedbits.net/debian-with-openbox-and-tint2-part-1-debian-network-install/) and made a nearly bare bones installation, with the intention of building everything after myself.
Will do the cleanup, thank you.

**Reply by tsujan** [reply]
I don't think it's caused by your Debian installation. I remembered that, years ago, I'd seen it in Debian but with another app. Didn't investigate why Debian was different from Arch in this regard.
